docs/migrations/support-slashed-denoms.md

```diff
@@ -0,0 +1,97 @@
+# Migrating from Unsupported Base Denom Slashes to Supporting Slashed BaseDenoms
+
+This document is intended to highlight significant changes which may require more information than presented in the CHANGELOG.
+Any changes that must be done by a user of ibc-go should be documented here.
+
+There are four sections based on the four potential user groups of this document:
+- Chains
+- IBC Apps
+- Relayers
+- IBC Light Clients
+
+**Note:** ibc-go supports golang semantic versioning and therefore all imports must be updated to bump the version number on major releases.
+```go
+github.com/cosmos/ibc-go/v3 -> github.com/cosmos/ibc-go/v4
+```
+
+This document is necessary when chains are upgrading from a version that does not support slashed base denoms (e.g. v3.0.0) to a version that does (e.g. v3.1.0).
+
+If a chain receives a slashed denom before it upgrades to supporting it, the receive may pass however the trace information will be incorrect.
+
+E.g. If a base denom of `testcoin/testcoin/testcoin` is sent to a chain that does not support slashes in the base denom; the receive will be successful. However, the trace information stored on the receiving chain will be: `Trace: "testcoin/testcoin", BaseDenom: "testcoin"`
+
+This incorrect trace information must be corrected when the chain does upgrade to fully supporting slashed denominations.
+
+To do so, chain binaries should include a migration script that will run when the chain upgrades from not supporting slashed base denominations to supporting slashed base denominations.
+
+## Chains
+
+### Transfer
+
+The transfer module will now support slashes in base denoms, so we must iterate over current traces to check if any of them are incorrectly formed and correct the trace information.
+
+### Upgrade Proposal
+
+```go
+// Here the upgrade name is the upgrade name set by the chain
+app.UpgradeKeeper.SetUpgradeHandler("supportSlashingDenomUpgrade",
+    func(ctx sdk.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+        // list of traces that must replace the old traces in store
+        var newTraces []transfertypes.DenomTrace
+
+        transferKeeper.IterateDenomTraces(ctx,
+        func(dt transfertypes.DenomTrace) bool {
+            // check if the new way of splitting FullDenom
+            // into Trace and BaseDenom is the same as the current
+            // DenomTrace.
+            // If it isn't then store the new DenomTrace in the list of new traces.
+            newTrace := transfertypes.ParseDenomTrace(dt.GetFullDenomPath())
+
+            if !reflect.DeepEqual(newTrace, dt) {
+                append(newTraces, newTrace)
+            }
+        })
+
+        // replace the outdated traces with the new trace information
+        for _, nt := range newTraces {
+            transferKeeper.SetDenomTrace(ctx, nt)
+        }
```

Comment:
Is it worth adding a call to [`Validate`](https://github.com/cosmos/ibc-go/blob/carlos/1446-bugtransfer-cannot-send-namespaced-ibc-denom-using-slashes/modules/apps/transfer/types/trace.go#L113) on new traces?

Reply:
Maybe we could add it in `if !reflect.DeepEqual(newTrace, dt)` as an extra condition. But if the validation fails, should we keep then  the existing denom trace?

Reply:
Yeah, I think we can add the `Validate` on that `if` condition and if it doesn't pass, then we just leave that denom trace as it is...